SmartSaver.MVC/Controllers/DashboardController.cs

```diff
@@ -39,7 +59,7 @@ public IActionResult Complete()
                 return View(nameof(Complete));
             }
 
-            return View(nameof(Index), account);
+            return RedirectToAction(nameof(Index), account);
```

Comment:
Just return View here. If another model is needed in the view, just redirect to Index action without any if statement. Index action will check that.